### DISCUSSION TITLE
How do I use Polyscale with Supabase from the Client side?

### DISCUSSION THREAD

**Discussion by suptejas** [supabase/supabase#11242] [Questions]
I have followed the integration article by Supabase detailing how to integrate Polyscale and Supabase.  I have created the Polyscale cache and I have my connection url.

However, when trying to access my database from the client side using the Supabase JS SDK, I have not been able to find a way to access my database through Supabase with my Polyscale connection URL, since I cannot specify it when I initialize the client.

How can I use Polyscale's database connection url with the Supabase client SDK? 

![image](https://user-images.githubusercontent.com/63039748/209431671-cd1164b8-4a18-4353-9495-bc7c1f16e034.png)

**Comment by haganbt** [comment]
Currently, PolyScale cannot be used with the Supabase JS client as it does not support a Supabase compatible HTTP api. You can however connect to your Supabase database via TCP with PolyScale.

**Reply by jetlej** [reply]
Major missed opportunity here :(

**Comment by DaleBeckles** [comment]
I went down the same rabbit hole as @suptejas except I was using the Dart Supabase client. The Supabase docs should explicitly state this limitation with PolyScale (and any others). I agree with @jetlej that this is a major missed opportunity - I was excited to give PolyScale a go.

@haganbt says that "currently" the Supabase client cannot be used with Polyscale. Are there any plans to address this, and if so, when?

**Comment by haganbt** [comment]
Hi @DaleBeckles - Apologies you ended up wasting your time. We will make sure the Supabase and PolyScale docs get updated to be explicit. 

We have had quite a few users asking for the same. We will be launching an HTTP API so as this gets fleshed out, this feedback is invaluable.

**Comment by christophemarois** [comment]
@haganbt also very interested in this, as it is the missing piece for making supabase globally available without having to spin up a backend.

**Comment by haganbt** [comment]
Thanks @christophemarois, we are starting our HTTP API implementation within the next week or so. Will update here once we have a beta available.

**Reply by chbert** [reply]
@haganbt any update on this? Would be great to use polyscale.

**Comment by haganbt** [comment]
HI @chbert - the good news is that we are launching our SQL over HTTP API very soon, the bad news is that this version is not Supabase compatible. You can read more here about the v1: https://docs.polyscale.ai/serverless-http-api
